Support analyzer: ^7 and source_gen: ^2 and Freezed  ^3

dorklein opened the pull request:
Update cloud_firestore_odm to support Freezed 3.0.0 and latest dependencies

- Resolve compatibility issues with Freezed 3.0.0 and above
- Update Dart SDK to >=3.0.0
- Upgrade dependencies to latest versions
- Add support for Freezed mixed mode classes
- Update generator to handle new Freezed class structures
- Improve compatibility with newer analyzer and source_gen versions

Fixes #39


> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

dorklein committed:
Update cloud_firestore_odm to support Freezed 3.0.0 and latest dependencies

- Resolve compatibility issues with Freezed 3.0.0 and above
- Update Dart SDK to >=3.0.0
- Upgrade dependencies to latest versions
- Add support for Freezed mixed mode classes
- Update generator to handle new Freezed class structures
- Improve compatibility with newer analyzer and source_gen versions

shtse8 commented:
anyone can merge ?

daviddomkar commented:
Hello, can this please be merged?

jannuskool commented:
Is there reason why this (and others too) merge is not happening? Has the project been abandoned?

mmatook commented:
Yup, not a lot of activity here. I use @dorklein s repo fork directly instead of this repo. Sad to see such a useful project abandoned. I understand that the original author is busy and has other commitments but for the sake of transparency please state so and give people a clear understanding what's going on. If ppl put the time in and contribute the least thing you can do is to merge or provide feedback.

jannuskool commented:
Could package updaters provide comment about status and path forward? @rrousselGit or someone else?

brendan-dexter commented:
@dorklein Is it possible to bump to the latest cloud_firestore?

dorklein committed:
chore: Update cloud_firestore_odm and generator to version 1.2.0

- Upgrade cloud_firestore dependency to ^6.0.0
- Update pubspec.yaml and CHANGELOG for versioning
- Modify transaction and batch methods to remove 'id' field from documents
- Enhance testing setup in README for running integration tests
- Add new tests for Freezed and other integration scenarios

dorklein commented:
> @dorklein Is it possible to bump to the latest cloud_firestore?

I updated to support cloud_firestore ^6.

The tests seem to pass, let me know if it works for you.

arun-prasad-r commented:
@rrousselGit @dorklein Any update on this?
